### PR TITLE
Report partial group connection failures through trace, not error

### PR DESCRIPTION
**Pull request by ollivir**
Also merges some code duplicated in FS4 / RPC invoker factories

Pairs with https://github.com/vespa-engine/system-test/pull/35

### CODE REVIEW

File: container-search/src/main/java/com/yahoo/search/dispatch/InterleavedSearchInvoker.java [hunk 1]
```diff
@@ -134,14 +136,34 @@ private void trimResult(Execution execution) {
...
+            }
             timedOut = true;
         }
+        if (alreadyFailedNodes != null) {
+            var message = "Connection failure on nodes with distribution-keys: "
+                    + alreadyFailedNodes.stream().map(v -> Integer.toString(v)).collect(Collectors.joining(", "));
+            if (asErrors) {
+                result.hits().addError(ErrorMessage.createBackendCommunicationError(message));
+            } else {
+                query.trace(message, 5);
```
Comment (bratseth) [inline container-search/src/main/java/com/yahoo/search/dispatch/InterleavedSearchInvoker.java]: (Same comment about level)

File: container-search/src/main/java/com/yahoo/search/dispatch/InterleavedSearchInvoker.java [hunk 2]
```diff
@@ -134,14 +136,34 @@ private void trimResult(Execution execution) {
...
         if (!invokers.isEmpty()) {
             String keys = invokers.stream().map(SearchInvoker::distributionKey).map(dk -> dk.map(i -> i.toString()).orElse("(unspecified)"))
                     .collect(Collectors.joining(", "));
 
-            result.hits().addError(ErrorMessage.createTimeout("Backend communication timeout on nodes with distribution-keys: " + keys));
+            if (asErrors) {
+                result.hits().addError(ErrorMessage
+                        .createTimeout("Backend communication timeout on all nodes in group (distribution-keys: " + keys + ")"));
+            } else {
+                query.trace("Backend communication timeout on nodes with distribution-keys: " + keys, 5);
```
[the remark this answers is not in the document]
Comment (bratseth) [reply container-search/src/main/java/com/yahoo/search/dispatch/InterleavedSearchInvoker.java]: I think we should have a metric for this as well.

**Review by bratseth — COMMENTED** [review]
otherwise LGTM!

File: container-search/src/main/java/com/yahoo/search/dispatch/InterleavedSearchInvoker.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ollivir) [reply container-search/src/main/java/com/yahoo/search/dispatch/InterleavedSearchInvoker.java]: Changed to 2 in new commit. I'll add the metrics as a separate sprint task

**Pull request merged**